File: rb/features.rb
# Bevwire SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module BevwireFeatures
  def self.make_feature(name)
    case name
    when "base"
      BevwireBaseFeature.new
    when "test"
      BevwireTestFeature.new
    else
      BevwireBaseFeature.new
    end
  end
end

File: py/features.py
# Bevwire SDK feature factory

from feature.base_feature import BevwireBaseFeature
from feature.test_feature import BevwireTestFeature


def _make_feature(name):
    features = {
        "base": lambda: BevwireBaseFeature(),
        "test": lambda: BevwireTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
